src/main/kotlin/no/nav/familie/ef/sak/beregning/BeregningUtils.kt

```diff
@@ -2,72 +2,98 @@ package no.nav.familie.ef.sak.beregning
+                          beregningsgrunnlag = Beregningsgrunnlag(samordningsfradrag = samordningsfradrag,
+                                                                  avkortningPerMåned = avkortningPerMåned,
+                                                                  fullOvergangsStønadPerMåned = fullOvergangsStønadPerMåned,
+                                                                  inntekt = inntekt,
+                                                                  grunnbeløp = it.beløp))
         }
+    }
+
+    private fun beregnAvkortning(grunnbeløp: BigDecimal, inntekt: BigDecimal): BigDecimal {
+        val inntektOverHalveGrunnbeløp = inntekt.subtract(grunnbeløp.multiply(BigDecimal(0.5)))
+        return if (inntektOverHalveGrunnbeløp > BigDecimal.ZERO)
+            inntektOverHalveGrunnbeløp.multiply(REDUKSJONSFAKTOR).setScale(5, RoundingMode.HALF_DOWN) else BigDecimal.ZERO
+    }
+
 
+    fun indeksjusterInntekt(sisteGrunnbeløpsdato: LocalDate,
+                            inntekter: List<Inntektsperiode> = emptyList()): List<Inntektsperiode> {
 
-        private fun beregnAvkortning(grunnbeløp: BigDecimal, inntekt: BigDecimal): BigDecimal {
-            val inntektOverHalveGrunnbeløp = inntekt.subtract(grunnbeløp.multiply(BigDecimal(0.5)))
-            return if (inntektOverHalveGrunnbeløp > BigDecimal.ZERO)
-                inntektOverHalveGrunnbeløp.multiply(REDUKSJONSFAKTOR).setScale(5, RoundingMode.HALF_DOWN) else BigDecimal.ZERO
+        val sistBrukteGrunnbeløp = finnGrunnbeløp(sisteGrunnbeløpsdato)
+        if (sisteGrunnbeløp == sistBrukteGrunnbeløp) {
+            return inntekter
         }
 
-        fun finnStartDatoOgSluttDatoForBeløpsperiode(beløpForInnteksperioder: List<Beløpsperiode>,
-                                                     vedtaksperiode: Periode): List<Beløpsperiode> {
-            return beløpForInnteksperioder.mapNotNull {
-                when {
-                    it.periode.omsluttesAv(vedtaksperiode) -> {
-                        it
-                    }
-                    it.periode.overlapperIStartenAv(vedtaksperiode) -> {
-                        it.copy(periode = it.periode.copy(fradato = vedtaksperiode.fradato))
-                    }
-                    vedtaksperiode.overlapperIStartenAv(it.periode) -> {
-                        it.copy(periode = it.periode.copy(tildato = vedtaksperiode.tildato))
-                    }
-                    vedtaksperiode.omsluttesAv(it.periode) -> {
-                        it.copy(periode = it.periode.copy(fradato = vedtaksperiode.fradato, tildato = vedtaksperiode.tildato))
-                    }
-                    else -> {
-                        null
-                    }
-                }
+        return inntekter.flatMap { justerInntektsperiode(it, sistBrukteGrunnbeløp) }
+
+    }
+
+    private fun justerInntektsperiode(inntektsperiode: Inntektsperiode,
```

Comment:
Jeg tolker denne som at man ønsker å lønnsjustere tvers ulike grunnbeløpsperioder, pga `finnGrunnbeløpsPerioder` og map, er det ønskelig? 
Trenger vi ikke kun å lønnsjustere de fra mai, og då egentlige bare burde ha en grunnbeløpsperiode?

Reply:
I utgangspunktet, men hvis det skulle vise seg at det blir behov for å juster g både i mai og juni, så er det jo veldig greit at koden støtter det tenker jeg.

Reply:
Vet vi at vi skal lønnsjustere vid begge tilfeller?

Reply:
Så lenge regelen er at vi skal inntekstsjustere, så bil det jo være riktig, og ja det er en test som tester det.